### PR TITLE
Block clone with namespace flag in seccomp default profile

### PR DESCRIPTION
**Pull request by PannagaRao**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

The default seccomp policy used by CRI-O did not previously block clone(2) system call (syscall) with CLONE_NEWNS and other namespace creation flags. Changes here block those flags. In addition, also disable the clone3(2) syscall as it is not currently possible to filter a struct argument (struct clone_args) with seccomp; this matches the approach containerd took.

#### Which issue(s) this PR fixes:

None

#### Special notes for your reviewer:

None

#### Does this PR introduce a user-facing change?

```release-note
The default seccomp policy now blocks clone and clone3 system calls that create a Linux namespace. This matches the default seccomp policy containerd uses.
```

### CODE REVIEW

**Comment by kwilczynski** [comment]
This change would break the ability to run anything within the container using Exec or ExecSync, or pretty much anything that will ever attempt to use fork()/vfork()/exec*()/posix_spawn(), anything backed by clone system call, that is.

**Comment by PannagaRao** [comment]
@dgl Could you please let me know if there were any checks to see if the implementation blocks only such clone() calls which have namespace flags as arguments.

File: test/testdata/clone-ns.c [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+#include <sched.h>
+
+int entry() {
+        system("id");
+        return 0;
+}
+
+int main() {
+        void *stack = malloc(1024*1024);
+        if (clone(entry, (stack+1024*1024), CLONE_NEWUSER|CLONE_NEWNET, 0) == -1) {
```
Comment (haircommander) [inline test/testdata/clone-ns.c:14]: can we toggle the clone flags on a cli flag so we can compress the two files and trigger each case from the same executable

**Comment by dgl** [comment]
[@kwilczynsk](https://github.com/cri-o/cri-o/pull/8437#issuecomment-2259764006):
> This change would break the ability to run anything within the container using Exec or ExecSync, or pretty much anything that will ever attempt to use fork()/vfork()/exec*()/posix_spawn(), anything backed by clone system call, that is.

The idea was it allows all clone args that aren't creating a namespace, then blocks the disallowed args (unless the container has CAP_SYS_ADMIN).

**Comment by kwilczynski** [comment]
Superseded by:

- https://github.com/cri-o/cri-o/pull/8514.

**Pull request closed without merging**